tests/unit/validation/test_validation_functions.py

```diff
@@ -135,10 +143,47 @@ def test_validate_ids_exist():
 
     update_data = {"source": source_update, "sym_load": sym_load_update}
 
-    invalid_ids = validate_ids_exist(update_data, input_data)
+    invalid_ids = validate_ids(update_data, input_data)
+
+    assert IdNotInDatasetError("source", [4], "update_data") in invalid_ids
+    assert IdNotInDatasetError("sym_load", [7], "update_data") in invalid_ids
+
+    source_update_no_id = initialize_array("update", "source", 3)
+    source_update_no_id["u_ref"] = [1.0, 2.0, 3.0]
+
+    update_data_col = compatibility_convert_row_columnar_dataset(
+        data={"source": source_update_no_id, "sym_load": sym_load_update},
+        data_filter=ComponentAttributeFilterOptions.relevant,
+        dataset_type=DatasetType.update,
+    )
+    invalid_ids = validate_ids(update_data_col, input_data)
+    assert len(invalid_ids) == 1
+    assert IdNotInDatasetError("sym_load", [7], "update_data") in invalid_ids
+
+    source_update_less_no_id = initialize_array("update", "source", 2)
+    source_update_less_no_id["u_ref"] = [1.0, 2.0]
 
-    assert IdNotInDatasetError("source", [4], "input_data") in invalid_ids
-    assert IdNotInDatasetError("sym_load", [7], "input_data") in invalid_ids
+    update_data_col_less_no_id = compatibility_convert_row_columnar_dataset(
+        data={"source": source_update_less_no_id, "sym_load": sym_load_update},
+        data_filter=ComponentAttributeFilterOptions.relevant,
+        dataset_type=DatasetType.update,
+    )
+    invalid_ids = validate_ids(update_data_col_less_no_id, input_data)
+    assert len(invalid_ids) == 2
+    assert IdNotInDatasetError("sym_load", [7], "update_data") in invalid_ids
+
+    source_update_part_nan_id = initialize_array("update", "source", 3)
+    source_update_part_nan_id["id"] = [1, np.iinfo(np.int32).min, 4]
```

Comment:
out of scope of this PR but we have the `nans` list in the `power_grid_meta_data`. maybe we should expose that (internally or externally) in a nicer way, if we can't use it here because it's a list and not a map